src/Common/MemoryTracker.cpp

```diff
@@ -133,17 +133,12 @@ void MemoryTracker::alloc(Int64 size)
         BlockerInThread untrack_lock;
 
         ProfileEvents::increment(ProfileEvents::QueryMemoryLimitExceeded);
-        std::stringstream message;
-        message.exceptions(std::ios::failbit);
-        message << "Memory tracker";
-        if (const auto * description = description_ptr.load(std::memory_order_relaxed))
-            message << " " << description;
-        message << ": fault injected. Would use " << formatReadableSizeWithBinarySuffix(will_be)
-            << " (attempt to allocate chunk of " << size << " bytes)"
-            << ", maximum: " << formatReadableSizeWithBinarySuffix(current_hard_limit);
-
+        const auto * description = description_ptr.load(std::memory_order_relaxed);
         amount.fetch_sub(size, std::memory_order_relaxed);
-        throw DB::Exception(message.str(), DB::ErrorCodes::MEMORY_LIMIT_EXCEEDED);
+        throw DB::Exception(DB::ErrorCodes::MEMORY_LIMIT_EXCEEDED,
+                            "Memory tracker {}: fault injected. Would use {} (attempt to allocate chunk of {} bytes), maximum: {}",
```

Comment:
Now it will have extra space if there is no description: `Memory tracker :` instead of `Memory tracker:`